src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +53,989 @@ const ActorInit Dm_Char08_InitVars = {
+        this->actionFunc = func_80AAFA18;
+    }
+    this->unk_1FA += 0xC8;
+    yOffset = 50.0f;
+    yOffset *= Math_SinS(this->unk_1FA);
+    Math_SmoothStepToF(&this->dyna.actor.world.pos.x, -6400.0f, 0.2f, 2.0f, 0.1f);
+    Math_SmoothStepToF(&this->targetYPos, -180.0f + yOffset, 0.5f, 5.0f, 0.1f);
+    Math_SmoothStepToF(&this->dyna.actor.world.pos.z, 1750.0f, 0.5f, 20.0f, 0.1f);
+    Math_SmoothStepToS(&this->dyna.actor.world.rot.y, -0x7234, 0xA, 0xDC, 1);
+    Math_SmoothStepToS(&this->dyna.actor.shape.rot.y, -0x7234, 0xA, 0xDC, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF15C.s")
+void func_80AAFA18(DmChar08* this, GlobalContext* globalCtx) {
+    s16 nextCs;
+    s16 nextCs2;
+    s16 nextCs1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/DmChar08_Init.s")
+    nextCs1 = ActorCutscene_GetAdditionalCutscene(this->dyna.actor.cutscene);
+    nextCs2 = nextCs1;
+    nextCs1 = ActorCutscene_GetAdditionalCutscene(nextCs1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/DmChar08_Destroy.s")
+    nextCs = ((void)0, gSaveContext.save.weekEventReg[53] & 0x40) ? nextCs1 : nextCs2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF610.s")
+    if (ActorCutscene_GetCanPlayNext(nextCs) != 0) {
+        gSaveContext.save.weekEventReg[53] |= 0x40;
+        ActorCutscene_Start(nextCs, &this->dyna.actor);
+        this->actionFunc = func_80AAFE78;
+    } else {
+        ActorCutscene_SetIntentToPlay(nextCs);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF79C.s")
+void func_80AAFAC4(DmChar08* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state == 0) {
+        this->actionFunc = func_80AAF8F4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF884.s")
+void func_80AAFAE4(DmChar08* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state == 0) {
+        this->actionFunc = func_80AAFB04;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAF8F4.s")
+void func_80AAFB04(DmChar08* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFA18.s")
+    if ((fabsf(this->dyna.actor.xzDistToPlayer) < 200.0f) && (player->actor.world.pos.y > 0.0f)) {
+        globalCtx->nextEntranceIndex = 0x6A70;
+        globalCtx->sceneLoadFlag = 0x14;
+        Scene_SetExitFade(globalCtx);
+        this->actionFunc = func_80AAFB94;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFAC4.s")
+void func_80AAFB94(DmChar08* this, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char08/func_80AAFAE4.s")
+Color_RGBA8 sBubblePrimColor = { 0xFA, 0xC8, 0xFA, 0x64 };
+Color_RGBA8 sBubbleEnvColor = { 0x50, 0x50, 0x50, 0 };
+Vec3f sBubbelAccel = { 0.0f, -0.8f, 0.0f };
```

Comment:
This names are a bit too generic. What about prefixing the actor name or similar?
Also, colors should be on dec
```suggestion
Color_RGBA8 sLargeTurtleBubblePrimColor = { 250, 200, 250, 100 };
Color_RGBA8 sLargeTurtleBubbleEnvColor = { 80, 80, 80, 0 };
Vec3f sLargeTurtleBubbelAccel = { 0.0f, -0.8f, 0.0f };
```